core/model/modx/processors/resource/trash/restore.class.php

```diff
@@ -0,0 +1,174 @@
+<?php
+
+/**
+ * Restores deleted files.
+ *
+ * @return boolean
+ * @package    modx
+ * @subpackage processors.resource
+ */
+class modResourceTrashRestoreProcessor extends modProcessor
+{
+
+    /** @var modResource $resource */
+    public $resource;
+
+    private $resourceList;
+    private $idList;
+    private $contexts;
+
+    public function getLanguageTopics()
+    {
+        return array('resource', 'trash');
+    }
+
+    /**
+     * @return bool|null|string
+     */
+    public function initialize()
+    {
+        // we expect a list of files here
+        $this->idList = explode(',', $this->getProperty('ids', false));
+        $this->modx->log(1, "Restorelist: " . print_r($this->idList, true));
+        $this->contexts = array();
+
+        if (count($this->idList)===0) {
+            return $this->modx->lexicon('resource_err_ns');
+        }
+
+        // and now retrieve a collection of resources here
+        $this->resourceList = $this->modx->getCollection('modResource', array(
+            'deleted' => true,
+            'id:IN'   => $this->idList,
+        ));
+        $count = sizeof($this->resourceList);
+
+        if (empty($this->resourceList)) {
+            return $this->modx->lexicon('resource_err_nfs', array('ids' => $this->idList));
+        }
+
+        $this->modx->log(1, "found " . $count);
+
+        foreach ($this->resourceList as $resource) {
+            $this->modx->log(1, "restoring " . $resource->get('id') . ": " . $resource->get('pagetitle'));
```

Comment:
as in the other place, probably better sent to the manager log instead of error log.